### PR TITLE
Fix evil clobbering active company-mode keymap

### LINKED ISSUE #664: `company-active-map` overridden when using company doc-buffer
When using company completion, if you open the doc buffer (`C-h`) then`company-active-map` will loose priority to `evil-insert-state-map`. This means that keys like `C-n`, `C-p`, and `C-k` no longer work for navigating the popup (interestingly `C-j`  still seems to work in this broken mode).

I was able to trace back this issue to this line in `company-doc-buffer`

https://github.com/company-mode/company-mode/blob/16ffeba5ef96c4c8e0cd39860b5402e25e304601/company.el#L2831

## to reproduce:
1. Begin company completion 
2.  `C-h` to open the doc buffer 
3. `C-n` will no longer work and will instead call `evil-complete-next` (The insert state binding)

### PR DESCRIPTION
**Pull request by CeleritasCelery**
fixes #664

- [ x ] byte-compiles cleanly
- [ x ] `M-x checkdoc` is happy.
- [ x ] define `evil-collection-mpc-setup` with `defun`
- [ x ] define `evil-collection-mpc-mode-maps` with `defconst`
- [ x ] All functions should start with `evil-collection-mpc-`

### CODE REVIEW

**Comment by jojojames** [comment]
This looks good. Would request a boolean to not turn off this workaround though (default it to enabled).

**Comment by CeleritasCelery** [comment]
I will add a boolean toggle. I am curious though, is there any reason someone would *not* want the fix enabled?

**Comment by condy0919** [comment]
> Asked another way, is it really worth adding a toggle for something that almost no one would want disabled? As opposed to just using the Emacs builtin way to turn it off?
> 
> ```lisp
> (remove-hook 'evil-local-mode-hook #'evil-collection-company-protect-keymap)
> ```

It's inconvenient for users.

**Comment by CeleritasCelery** [comment]
Thanks. Added the toggle.

**Comment by condy0919** [comment]
> Thanks. Added the toggle.

squash your commits. The commit message should be like "company-mode: ..." or something else.

**Comment by CeleritasCelery** [comment]
done

File: modes/company/evil-collection-company.el [hunk 1]
```diff
@@ -112,6 +126,9 @@ C-x C-l."
     (kbd "M-k") 'company-select-previous
     (kbd "<escape>") 'company-search-abort)
 
+  ;; https://github.com/emacs-evil/evil-collection/issues/664
+  (add-hook 'evil-local-mode-hook #'evil-collection-company-protect-keymap)
```
Comment (jojojames) [inline modes/company/evil-collection-company.el:130]: I'll probably move the toggle around this add-hook instead at some point if I remember. Thanks!

**Pull request merged**